selfdrive/sensord/sensors_qcom.cc

```diff
@@ -70,30 +69,33 @@ void sensor_loop() {
       LOGD("sensor %4d: %4d %60s  %d-%ld us", i, list[i].handle, list[i].name, list[i].minDelay, list[i].maxDelay);
     }
 
-    device->activate(device, SENSOR_MAGNETOMETER_UNCALIBRATED, 0);
-    device->activate(device, SENSOR_GYRO_UNCALIBRATED, 0);
-    device->activate(device, SENSOR_ACCELEROMETER, 0);
-    device->activate(device, SENSOR_MAGNETOMETER, 0);
-    device->activate(device, SENSOR_GYRO, 0);
-    device->activate(device, SENSOR_PROXIMITY, 0);
-    device->activate(device, SENSOR_LIGHT, 0);
-
-    device->activate(device, SENSOR_MAGNETOMETER_UNCALIBRATED, 1);
-    device->activate(device, SENSOR_GYRO_UNCALIBRATED, 1);
-    device->activate(device, SENSOR_ACCELEROMETER, 1);
-    device->activate(device, SENSOR_MAGNETOMETER, 1);
-    device->activate(device, SENSOR_GYRO, 1);
-    device->activate(device, SENSOR_PROXIMITY, 1);
-    device->activate(device, SENSOR_LIGHT, 1);
-
-    device->setDelay(device, SENSOR_GYRO_UNCALIBRATED, ms2ns(10));
-    device->setDelay(device, SENSOR_MAGNETOMETER_UNCALIBRATED, ms2ns(100));
-    device->setDelay(device, SENSOR_ACCELEROMETER, ms2ns(10));
-    device->setDelay(device, SENSOR_GYRO, ms2ns(10));
-    device->setDelay(device, SENSOR_MAGNETOMETER, ms2ns(100));
-    device->setDelay(device, SENSOR_PROXIMITY, ms2ns(100));
-    device->setDelay(device, SENSOR_LIGHT, ms2ns(100));
+    std::set<int> sensor_types = {
+      SENSOR_TYPE_ACCELEROMETER,
+      SENSOR_TYPE_MAGNETIC_FIELD_UNCALIBRATED,
+      SENSOR_TYPE_MAGNETIC_FIELD,
+      SENSOR_TYPE_GYROSCOPE_UNCALIBRATED,
+      SENSOR_TYPE_GYROSCOPE,
+      SENSOR_TYPE_PROXIMITY,
+      SENSOR_TYPE_LIGHT,
+    };
+
+    std::map<int, int64_t> sensors = {
+      {SENSOR_GYRO_UNCALIBRATED, ms2ns(10)},
+      {SENSOR_MAGNETOMETER_UNCALIBRATED, ms2ns(100)},
+      {SENSOR_ACCELEROMETER, ms2ns(10)},
+      {SENSOR_GYRO, ms2ns(10)},
+      {SENSOR_MAGNETOMETER, ms2ns(100)},
+      {SENSOR_PROXIMITY, ms2ns(100)},
+      {SENSOR_LIGHT, ms2ns(100)}
+    };
```

Comment:
It might be simpler to use a sensor struct?

Reply:
think I'll just merge as is since it'll need to be revisited at some point. right now, we have the qualcomm daemon and our daemon running (4% and 6% CPU respectively), and ideally we should be able to only use one daemon with far less CPU

Reply:
oh I get it.